core/src/main/java/org/opensearch/sql/expression/function/OpenSearchFunctions.java

```diff
@@ -86,6 +89,36 @@ private static FunctionResolver wildcard_query(BuiltinFunctionName wildcardQuery
     return new RelevanceFunctionResolver(funcName);
   }
 
+  private static FunctionResolver nested() {
+    return new FunctionResolver() {
+      @Override
+      public Pair<FunctionSignature, FunctionBuilder> resolve(
+          FunctionSignature unresolvedSignature) {
+        return Pair.of(unresolvedSignature,
+            (functionProperties, arguments) ->
+            new FunctionExpression(BuiltinFunctionName.NESTED.getName(), arguments) {
+              @Override
+              public ExprValue valueOf(Environment<Expression, ExprValue> valueEnv) {
+                return valueEnv.resolve(getArguments().get(0));
```

Comment:
I did semantic checks in [NestedAnalyzer](https://github.com/opensearch-project/sql/blob/3ff9fdd1cce7fdcf78281720bb85d62bd6c2a8dd/core/src/main/java/org/opensearch/sql/analysis/NestedAnalyzer.java#L82-L101).

Reply:
I see. Could you confirm if we need to move `validateArgs()` logic here? I recall we do argument check either automatically by providing function signature or doing it in function resolver.

Reply:
We do need to do semantic checks in the `NestedAnalyzer` to ensure the LogicalPlan is fulfilled correctly for push down. I avoided the function signature because the param and return types can be any, and we only need to resolve the first argument while using the path for push down and flattening logic. Does this implementation make sense to you? I think this is a simple way to do the semantic checks.

Reply:
Okay, I see. Thanks!